### PR TITLE
Documentation for IAlterSuiteListener

### PR DESCRIPTION
**Pull request by krmahadevan**

### CODE REVIEW

**Comment by krmahadevan** [comment]
@cbeust / @juherr 
This PR adds documentation or https://github.com/cbeust/testng/pull/873
Changes to Changes.txt were already made as part of the earlier PR [ https://github.com/cbeust/testng/blob/master/CHANGES.txt#L4 ]

**Comment by juherr** [comment]
:+1:

File: doc/documentation-main.html [hunk 1]
```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
 }
 </pre>
 
+<p>
+<!-------------------------------------
+  IALTERSUITELISTENER
+  ------------------------------------>
+
+<h4><a class="section" indent=".." name="ialtersuite">Altering suites (or) tests</a></h4>
+Sometimes we may need to just want to alter a suite (or) a test tag in a suite xml in runtime without having to
```
Comment (cbeust) [inline doc/documentation-main.html]: you may need

File: doc/documentation-main.html [hunk 2]
```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
...
+change the contents of a suite file.<br>
```
Comment (cbeust) [inline doc/documentation-main.html]: Use `<p>` for paragraphs.

File: doc/documentation-main.html [hunk 3]
```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
...
+
+A classic example for this would be to try and leverage your existing suite file and try using it for simulating a load test
+on your "Application under test".
+If you were to do this on your own, at the minimum you would end up duplicating the contents of your &lt;test&gt; tag multiple
```
Comment (cbeust) [inline doc/documentation-main.html]: This sentence "if you were to do this`" is unnecessary, remove it.

File: doc/documentation-main.html [hunk 4]
```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
...
+times and create a new suite xml file and work with. But this doesn't seem to scale a lot.
+
```
Comment (cbeust) [inline doc/documentation-main.html]: `<p>`

File: doc/documentation-main.html [hunk 5]
```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
...
+    @Override
+    public void alter(List&lt;XmlSuite&gt; suites) {
+        XmlSuite suite = suites.get(0);
+        suite.setName(getClass().getSimpleName());
+    }
+}
+</pre>
+
+This listener can only be added to your execution via the following two ways :
```
Comment (cbeust) [inline doc/documentation-main.html]: "This listener can only be added with either of the following ways" (your phrasing makes it sound you need to do both).

File: doc/documentation-main.html [hunk 6]
```diff
@@ -2343,6 +2343,45 @@ <h3 class="sourcetitle">Shell</h3>
...
+<ul>
+  <li>Through the <tt>&lt;listeners&gt;</tt> tag in the suite xml file.</li>
+  <li>Through the mechanism of <a href="#listeners-service-loader">ServiceLoaders</a></li>
```
Comment (cbeust) [inline doc/documentation-main.html]:
```
Through a  <a href="#listeners-service-loader">service loader</a></li>
```

File: doc/documentation-main.html (hunk 1 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: fixed

File: doc/documentation-main.html (hunk 2 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: fixed

File: doc/documentation-main.html (hunk 3 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: fixed

File: doc/documentation-main.html (hunk 4 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: fixed

File: doc/documentation-main.html (hunk 5 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: fixed

File: doc/documentation-main.html (hunk 6 above)
Comment (krmahadevan) [inline doc/documentation-main.html]: fixed

**Comment by krmahadevan** [comment]
@cbeust 
Thanks for helping vet out the documentation content. I have addressed all your comments.

**Pull request merged**